src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -155,6 +180,12 @@ private async Task<RazorCodeDocument> GetRazorCodeDocumentAsync(DocumentSnapshot
 
         private IReadOnlyList<SemanticRange> CombineSemanticRanges(params IReadOnlyList<SemanticRange>[] rangesArray)
         {
+            if (rangesArray.Any(a => a is null))
```

Comment:
I'm confused, when would an individual range be `null`? I'd imagine an entire C# semantic token response being `null` because of version mismatch but not a range.

Reply:
Hmm, maybe rangesArray is a bad name. This method takes `IReadOnlyList<SemanticRange>[]`, so my `.Any` is over each `IReadOnlyList<SemanticRange>`, which is "an entire semantic token response". I'm open to better names, I'm coming up short ATM.

Reply:
ahhhh, so we throw away our entire semantic model if one of the ranges are invalid?Couldn't we just re-use the pre-existing C# semantic ranges?

Reply:
We throw away our entire semantic model if C# doesn't give us a response/that response is not in sync with us. If we re-used existing C# semantic ranges then they would be of a different documentVersion than the one we're using for Razor and it could have the same problems with overlapping and misalignment that we currently experience.

Thankfully the behavior when our LSP returns `null` is to change nothing, which means that this results in a delay of update, but no flashing (which is what we get if we return only razor results, which was my original plan).

Reply:
Ahhh that totally makes sense. Thanks for the clarification!